### ISSUE TITLE
Condition number is computed wrong in some cases

### ISSUE THREAD

**Issue by goggle** [JuliaLang/julia#33547]
I have discovered this issue while working on another one:

```julia
julia> M = [1.0 -2.0; -2.0 -1.5]
2×2 Array{Float64,2}:
  1.0  -2.0
 -2.0  -1.5

julia> S = Symmetric(M)
2×2 Symmetric{Float64,Array{Float64,2}}:
  1.0  -2.0
 -2.0  -1.5

julia> cond(M, 1)
1.909090909090909

julia> cond(S, 1)
2.227272727272727
```

These two condition numbers should be the same, but they are clearly not.

**Comment by antoine-levitt** [comment]
The issue is that `_cond1Inf` in `lu.jl` calls `gecon`, which is only an estimate of the matrix norm, so `S` is actually the correct one. Let's just remove the specialization and just compute `opnorm(A, p)*opnorm(inv(A), p)` in all cases? Condition number is only a diagnostic tool, so its speed is not that important. Affects https://github.com/JuliaLang/julia/issues/33297.

**Comment by goggle** [comment]
@antoine-levitt Yes exactly, I was preparing a PR for #33297, when this issue appeared after writing some tests for it.

Yes, I think removing the specialization
```julia
_cond1Inf(A::StridedMatrix{<:BlasFloat}, p::Real)
```
would resolve this.

**Comment by antoine-levitt** [comment]
I went ahead and did a PR for this, if you have any additional fixes or tests feel free to add them
